src/BlobStore.hs

```diff
@@ -161,30 +154,27 @@ instance BlobStore BlobDirStore where
   getBlob bd (ExtantBlob name) = sourceFile (blobFileName bd name)
 
 blobFileName :: BlobDirStore -> BS.ByteString -> FilePath
-blobFileName (BlobDir d) blobname = d </> toString (Base64.encode blobname)
+blobFileName (BlobDir d) = (d </>) . toString . Base64.encode
 
 unBlobFileName :: FilePath -> Maybe BS.ByteString
-unBlobFileName relpath =
-  case Base64.decode $ fromString relpath of
-    Left _ -> Nothing
-    Right n ->
-      if BS.length n == blobNameLength
-        then Just n
-        else Nothing
-
-sinkTrustedBlob ::
-     (BlobStore bs, MonadResource m)
+unBlobFileName
+  = find ((== blobNameLength) . BS.length)
+  . either (const Nothing) Just
+  . Base64.decode
+  . fromString
```

Comment:
When if first looked at this, the `Left _ -> Nothing` got my attention. I personally get a code smell when I see  simple pattern matches that are dropping their values, as well as simple cases on `Either` or `Maybe` values. This got me thinking about maybe using the `either` combinator. Additionally when using `if` to eventually return a `Maybe`, if often think using `find :: Foldable t => (a -> Bool) -> t a -> Maybe a`. 

Given that I like like composition pipelines, I like my refactor. But overall I think this is mostly a nit and personal preference.